pkg/operation/operation.go

```diff
@@ -608,6 +609,16 @@ func (o *Operation) ComputeLokiHosts() []string {
 	}
 }
 
+// IsShootMonitoringEnabled returns true if shoot monitoring is enabled and shoot is not of purpose testing.
+func (o *Operation) IsShootMonitoringEnabled() bool {
+	return helper.IsMonitoringEnabled(o.Config) && o.Shoot.Purpose != gardencorev1beta1.ShootPurposeTesting
+}
+
+// WantsGrafana returns true if shoot is not of purpose testing and either shoot monitoring or loki is enabled.
+func (o *Operation) WantsGrafana() bool {
+	return o.Shoot.Purpose != gardencorev1beta1.ShootPurposeTesting && (helper.IsMonitoringEnabled(o.Config) || helper.IsLokiEnabled(o.Config))
+}
+
```

Comment:
You could add unit tests.

Reply:
No. I'm not gonna unit test a boolean check. The helper functions are already tested.